tests/testthat/test-pstr_add_reductions.R

```diff
@@ -1,71 +1,82 @@
-test_that("throws an error when the user do not give enough parameters", {
-  expect_error(pstr_add_reductions())
-  expect_error(pstr_add_reductions(pstr_toy_companies()))
-  expect_error(pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo()))
-})
+test_that("with missing obligatory arguments errors gracefully", {
+  companies <- pstr_toy_companies()
+  ep_weo <- pstr_toy_ep_weo()
+  weo <- pstr_toy_weo_2022()
 
-test_that("throws an error when the user do not give the parameters in the right order", {
-  expect_error(pstr_add_reductions(pstr_toy_ep_weo(), pstr_toy_companies(), pstr_toy_weo_2022()))
-  expect_error(pstr_add_reductions(pstr_toy_weo_2022(), pstr_toy_companies(), pstr_toy_ep_weo()))
+  expect_error(pstr_add_reductions(companies, ep_weo = ep_weo), "weo_2022")
+  expect_error(pstr_add_reductions(companies, weo_2022 = weo_2022), "ep_weo")
+  expect_error(pstr_add_reductions(ep_weo = ep_weo, weo = weo), "companies")
 })
 
-test_that("returns a tibble data frame", {
-  expect_s3_class(
-    pstr_add_reductions(pstr_toy_companies(), pstr_toy_ep_weo(), pstr_toy_weo_2022()),
-    "tbl_df"
-  )
+test_that("returns a tibble", {
+  companies <- pstr_toy_companies()
+  ep_weo <- pstr_toy_ep_weo()
+  weo <- pstr_toy_weo_2022()
+
+  out <- pstr_add_reductions(companies, ep_weo, weo)
+
+  expect_s3_class(out, "tbl_df")
```

Comment:
The "code paragraphs" here follow a popular pattern called AAA. Google "aaa in tests". Note how the lines got a lot shorter by using meaningful names.